pacman/utilities/utility_objs/resource_tracker.py

```diff
@@ -668,21 +731,25 @@ def _allocate_core(self, chip, key, processor_id):
         :type processor_id: int
         """
         if key not in self._core_tracker:
-            self._core_tracker[key] = set()
-            for processor in chip.processors:
-                if not processor.is_monitor:
-                    self._core_tracker[key].add(processor.processor_id)
+            self._fill_in_core_tracker_for_chip(key, chip)
         if processor_id is not None:
             self._core_tracker[key].remove(processor_id)
         else:
 
             # TODO: Find a core that meets the resource requirements
             processor_id = self._core_tracker[key].pop()
 
-        if len(self._core_tracker[key]) == 0:
+        if (len(self._core_tracker[key]) ==
+                0 + self._previous_allocated_resources[key]):
```

Comment:
could just be == self._previous_allocated_resources[key]):

post commit modification.